File: 2econd/C++/main.cpp
#define _USE_MATH_DEFINES
#include <math.h>
#include <iostream>
class Figure {
public:
    virtual double area() {return 0.0;}
    virtual double perimeter() {return 0.0;}
};
class Rectangle : public Figure {
public:
    double ax, ay, bx, by, cx, cy, dx, dy, horizontal, vertical;
    Rectangle (double ax, double ay, double cx, double cy) {
        this->ax = ax;
        this->ay = ay;
        this->bx = ax;
        this->by = cy;
        this->cx = cx;
        this->cy = cy;
        this->dx = cx;
        this->dy = ay;
        this->horizontal = cx - ax;
        this->vertical = cy - ay;
    }
    double area() {
        return this->horizontal * this->vertical;
    }
    double perimeter() {
        return 2.0 * (this->vertical + this->horizontal);
    }
};
class Circle : public Figure {
public:
    double radius;
    Circle (double radius) { this->radius = radius;}
    double area() {
        return M_PI * pow(this->radius, 2.0);
    }
    double perimeter() {
        return 2.00 * M_PI * this->radius;
    }
};
int main() {
    std::cout << ("Так розташовані вугли прямокутника\nB--------C\n|        |\n|        |\nA--------D\n") << std::endl;
    double ax = 0.0, ay = 0.0, cx = 40.0, cy = 15.0;
    Rectangle rectangle(ax, ay, cx, cy);
    std::cout << "A(x) = " << ax <<"\nA(y) = " << ay << "\nC(x) = " << cx << "\nC(y) = " << cy << std::endl;
    std::cout << "Площа прямокутника = " << rectangle.area() << std::endl;
    std::cout << "Периметр прямокутника = " << rectangle.perimeter() << std::endl << std::endl;

    double radius = 18.00;
    Circle circle(radius);
    std::cout << "Площа кругу = " << circle.area() <<std::endl;
    std::cout << "Периметр кругу = " << circle.perimeter() <<std::endl;
    return 0;
}

File: 2econd/Python/main.py
from math import pi


class Figure:
    def area(self):
        pass

    def perimeter(self):
        pass


class Rectangle(Figure):
    def __init__(self, ax, ay, cx, cy):
        self.ax = ax
        self.ay = ay
        self.bx = ax
        self.by = cy
        self.cx = cx
        self.cy = cy
        self.dx = cx
        self.dy = ay
        self.horizontal = cx - ax
        self.vertical = cy - ay

    def area(self):
        return self.horizontal * self.vertical

    def perimeter(self):
        return 2*(self.horizontal+self.vertical)


class Circle(Figure):
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return pi * (self.radius ** 2)

    def perimeter(self):
        return 2 * pi * self.radius


if __name__ == '__main__':
    print("""Так розташовані вугли прямокутника
    B--------C
    |        |
    |        |
    A--------D""")

    ax = 0.0
    ay = 0.0
    cx = 40.0
    cy = 15.0
    rectangle = Rectangle(ax, ay, cx, cy)
    print(f"A(x) = {ax}\n"
          f"A(y) = {ay}\n"
          f"C(x) = {cx}\n"
          f"C(y) = {cy}")
    print(f"Площа прямокутника = {round(rectangle.area(), 4)}")
    print(f"Периметр прямокутника = {round(rectangle.perimeter(), 4)}\n")

    radius = 18.00
    circle = Circle(radius)
    print(f"Площа кругу = {round(circle.area(), 4)}")
    print(f"Периметр кругу = {round(circle.perimeter(), 4)}")